networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/Eth2TopicHandler.java

```diff
@@ -62,15 +62,20 @@ protected abstract void processMessage(
       final TWrapped message, InternalValidationResult internalValidationResult);
 
   protected ValidationResult handleMessageProcessingError(Throwable err) {
+    final ValidationResult response;
     if (Throwables.getRootCause(err) instanceof DecodingException) {
       LOG.trace("Received malformed gossip message on {}", getTopic());
+      response = ValidationResult.Invalid;
     } else if (Throwables.getRootCause(err) instanceof RejectedExecutionException) {
       LOG.warn(
           "Discarding gossip message for topic {} because the executor queue is full", getTopic());
+      response = ValidationResult.Ignore;
     } else {
       LOG.warn("Encountered exception while processing message for topic {}", getTopic(), err);
+      response = ValidationResult.Ignore;
```

Comment:
From my perspective I would `Ignore` only the cases when we are 100% sure this is not the remote peer guilt. 
I'm almost sure there are still cases when other exceptions could be thrown beside `DecodingException` and this could be the attack target and in this case the peer would not be banned. IMHO it's better to unfairly ban a honest peer than keep being DoSed by a malicious peer.

Reply:
Yeah, on second thought this makes a lot of sense 👍  And we do log a warning, so we should be able to detect any cases where we're being too strict.